core/DataTable.php

```diff
@@ -591,9 +591,6 @@ public function addDataTable(DataTable $tableToSum)
     public function getRowFromLabel($label)
     {
         $rowId = $this->getRowIdFromLabel($label);
-        if ($rowId instanceof Row) {
```

Comment:
`$rowId` usually always returns an integer. I don't think this method should ever return a `Row` at all but not sure. Anyhow, we move this check to the end which improves performance. When archiving a year this method was called 1 million times (on a bit bigger instance) so it does make quite a difference.